index.js

```diff
@@ -186,6 +190,28 @@ class LedgerBridgeKeyring extends EventEmitter {
     delete this.accountDetails[ethUtil.toChecksumAddress(address)]
   }
 
+  updateTransportMethod(useLedgerLive = false) {
+    return new Promise((resolve, reject) => {
+      // If the iframe isn't loaded yet, let's store the desired useLedgerLive value and
+      // optimistically return a successful promise
+      if(!this.iframeLoaded) {
+        this.initialUseLedgerLive = useLedgerLive
+        return Promise.resolve(true);
```

Comment:
This will effectively hide the error in this case.

Branching here if the iframe hasn't load yet makes sense, but could we return the Promise result on like 335 instead? Then this the caller would know when the transport method was set, and would know when it failed.

This could be done by constructing a Promise, attaching it to `this`, and returning it without resolving it. Then it can be resolved below in `onload`.